File: src/main/java/com/nethsoft/core/util/DateUtil.java
package com.nethsoft.core.util;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

public class DateUtil {
	public static Date parseDate(String dateStr, String format) {
		Date date = null;
		try {
			DateFormat dateFormat = new SimpleDateFormat(format);
			String dt = dateStr.replaceAll("-", "-");
			if ((!dt.equals("")) && (dt.length() < format.length())) {
				dt = dt
						+ format.substring(dt.length()).replaceAll(
								"[YyMmDdHhSs]", "0");
			}
			date = dateFormat.parse(dt);
		} catch (Exception e) {
			e.printStackTrace();
		}
		return date;
	}

	public static Date parseDate(String dateStr) {
		return parseDate(dateStr, "yyyy-MM-dd");
	}

	public static Date parseDateTime(String dateStr) {
		return parseDate(dateStr, "yyyy-MM-dd HH:mm:ss");
	}

	public static String format(Date date, String format) {
		String result = "";
		try {
			if (date != null) {
				DateFormat dateFormat = new SimpleDateFormat(format);
				result = dateFormat.format(date);
			}
		} catch (Exception localException) {
		}
		return result;
	}

	public static String format(Date date) {
		return format(date, "yyyy-MM-dd");
	}

	public static int getYear(Date date) {
		Calendar calendar = Calendar.getInstance();
		calendar.setTime(date);
		return calendar.get(1);
	}

	public static int getMonth(Date date) {
		Calendar calendar = Calendar.getInstance();
		calendar.setTime(date);
		return calendar.get(2) + 1;
	}

	public static int getDay(Date date) {
		Calendar calendar = Calendar.getInstance();
		calendar.setTime(date);
		return calendar.get(5);
	}

	public static int getHour(Date date) {
		Calendar calendar = Calendar.getInstance();
		calendar.setTime(date);
		return calendar.get(11);
	}

	public static int getMinute(Date date) {
		Calendar calendar = Calendar.getInstance();
		calendar.setTime(date);
		return calendar.get(12);
	}

	public static int getSecond(Date date) {
		Calendar calendar = Calendar.getInstance();
		calendar.setTime(date);
		return calendar.get(13);
	}

	public static long getMillis(Date date) {
		Calendar calendar = Calendar.getInstance();
		calendar.setTime(date);
		return calendar.getTimeInMillis();
	}

	public static String getDate(Date date) {
		return format(date, "yyyy-MM-dd");
	}

	public static String getTime(Date date) {
		return format(date, "HH:mm:ss");
	}
	/**
	 * 获取当前时间
	 * @return
	 */
	public static String now(){
		return getDateTime(new Date());
	}
	/**
	 * 获取当前时间
	 * @param 格式
	 * @return
	 */
	public static String now(String format){
		return format(new Date(), format);
	}
	/**
	 * 格式 yyyy-MM-dd HH:mm:ss
	 * @param date
	 * @return
	 */
	public static String getDateTime(Date date) {
		return format(date, "yyyy-MM-dd HH:mm:ss");
	}

	public static Date addDate(Date date, int day) {
		Calendar calendar = Calendar.getInstance();
		long millis = getMillis(date) + day * 24L * 3600L * 1000L;
		calendar.setTimeInMillis(millis);
		return calendar.getTime();
	}

	public static Date minDate(Date date, int day) {
		Calendar calendar = Calendar.getInstance();
		long millis = getMillis(date) - day * 24L * 3600L * 1000L;
		calendar.setTimeInMillis(millis);
		return calendar.getTime();
	}

	public static int diffDate(Date date, Date date1) {
		return (int) ((getMillis(date) - getMillis(date1)) / 86400000L);
	}

	public static String getMonthBegin(String strdate) {
		Date date = parseDate(strdate);
		return format(date, "yyyy-MM") + "-01";
	}

	public static String getMonthEnd(String strdate) {
		Date date = parseDate(getMonthBegin(strdate));
		Calendar calendar = Calendar.getInstance();
		calendar.setTime(date);
		calendar.add(2, 1);
		calendar.add(6, -1);
		return formatDate(calendar.getTime());
	}

	public static String formatDate(Date date) {
		return formatDateByFormat(date, "yyyy-MM-dd");
	}

	public static String formatDateByFormat(Date date, String format) {
		String result = "";
		if (date != null) {
			try {
				SimpleDateFormat sdf = new SimpleDateFormat(format);
				result = sdf.format(date);
			} catch (Exception ex) {
				ex.printStackTrace();
			}
		}
		return result;
	}
/**
 * 获取星期
 * @return
 */
	public static String getWeek() {
		int week = Calendar.getInstance().get(Calendar.DAY_OF_WEEK);
		String wk = "";
		switch(week){
		case 1:
			wk = "星期天";
			break;
		case 2:
			wk = "星期一";
			break;
		case 3:
			wk = "星期二";
			break;
		case 4:
			wk = "星期三";
			break;
		case 5:
			wk = "星期四";
			break;
		case 6:
			wk = "星期五";
			break;
		case 7:
			wk = "星期六";
			break;
		}
		return wk;
	}

	/**
	 * 根据当前日期获取多少天之后的日期
	 * @param num
	 * @return
     */
	public static Date dateAdd(int num){
		Calendar calendar =  Calendar.getInstance();
		calendar.add(5,num);
		Date date = calendar.getTime();
		return date;
	}

	/**
	 * 根据日期获取多少天之后的日期
	 * @param num
	 * @return
	 */
	public static Date dateAdd(Date date,int num){
		Calendar calendar =  Calendar.getInstance();
		calendar.setTime(date);
		calendar.add(5,num);
		return calendar.getTime();
	}
}

File: src/main/webapp/resources/framework/js/plugins/util/DateUtil.js

/**
 * 格式化【日期】
 * @param date
 * @returns {string}
 */
function formatDate(date){
    var year = date.getFullYear();
    var month = date.getMonth()+1;
    var day = date.getDate();
    return year+"-"+(month<10?"0"+month:month)+"-"+(day<10?"0"+day:day);
}

var DateUtil = {
    format:function(o){
        var s = "";
        if(typeof(o) == "string"){
            s = o;
        }else{
            s = o.value;
        }
        if(s.length == 4){
            o.value = s+"-";
        }
        if(s.length == 7){
            o.value = s+"-";
        }
        if(s.length > 10){
            o.value = s.substring(0,10);
        }
    },
    lastweek : function(){
        var date = new Date(new Date().getTime() - 1000 * 60 * 60 * 24 * 6);
        return formatDate(date);
    },
    daysInMonth : [0,31,28,31,30,31,30,31,31,30,31,30,31],
    lastmonth : function(){
        var now = new Date();
        var daysInMonth = this.daysInMonth;
        var strYear = now.getFullYear();
        var strDay = now.getDate();
        var strMonth = now.getMonth() + 1;
        if (strYear % 4 == 0 && strYear % 100 != 0) {
            daysInMonth[2] = 29;
        }
        if (strMonth - 1 == 0) {
            strYear -= 1;
            strMonth = 12;
        }
        else {
            strMonth -= 1;
        }
        strDay = daysInMonth[strMonth] >= strDay ? strDay : daysInMonth[strMonth];
        if (strMonth < 10) {
            strMonth = "0" + strMonth;
        }
        if (strDay < 10) {
            strDay = "0" + strDay;
        }
        return strYear + "-" + strMonth + "-" + strDay;
    },
    last3month : function(){
        var now = new Date();
        var daysInMonth = this.daysInMonth;
        var strYear = now.getFullYear();
        var strDay = now.getDate();
        var strMonth = now.getMonth() + 1;
        if (strYear % 4 == 0 && strYear % 100 != 0) {
            daysInMonth[2] = 29;
        }
        if (strMonth - 3 > 0) {
            strMonth -= 3;
        }else if(strMonth -1 == 0){
            strYear -= 1;
            strMonth = 10;
        }else if(strMonth -2 == 0){
            strYear -= 1;
            strMonth = 11;
        }else{
            strYear -= 1;
            strMonth = 12;
        }
        strDay = daysInMonth[strMonth] >= strDay ? strDay : daysInMonth[strMonth];
        if (strMonth < 10) {
            strMonth = "0" + strMonth;
        }
        if (strDay < 10) {
            strDay = "0" + strDay;
        }
        return strYear + "-" + strMonth + "-" + strDay;
    },
    lastyear : function(){
        var now = new Date();
        var strYear = now.getFullYear() - 1;
        var strDay = now.getDate();
        var strMonth = now.getMonth()+1;
        if(strMonth<10){
            strMonth="0"+strMonth;
        }
        if(strDay<10){
            strDay="0"+strDay;
        }
        return strYear+"-"+strMonth+"-"+strDay;
    }
};

